pkg/compose/build.go

```diff
@@ -61,55 +65,37 @@ func (s *composeService) build(ctx context.Context, project *types.Project, opti
 			if service.Name != name {
 				continue
 			}
-			service, err := project.GetService(name)
-			if err != nil {
-				return err
-			}
+
 			if service.Build == nil {
 				return nil
 			}
-			imageName := api.GetImageNameOrDefault(service, project.Name)
-			buildOptions, err := s.toBuildOptions(project, service, imageName, options.SSHs)
-			if err != nil {
-				return err
+
+			if buildkitEnabled, err := s.dockerCli.BuildKitEnabled(); err != nil || !buildkitEnabled {
```

Comment:
Minor: it might be good to bring this out of the "loop" (`InDependencyOrder`)

Also, I think it's probably safe to fail and propagate the error:
https://github.com/docker/cli/blob/3c9e0073dd3ab31783a9d27d5fc26b70d59db367/cli/command/cli.go#L171 (tl;dr it can only fail if you do something like `DOCKER_BUILDKIT=not_a_bool`

Or, at the very least, a warning that we're falling back would be nice.